File: leetcode/0041.First-Missing-Positive/FirstMissingPositive.py
from typing import List

class Solution:

    # Runtime: 32 ms, faster than 90.29% of Python3 online submissions for First Missing Positive.
    def firstMissingPositive(self, nums: List[int]) -> int:
        nums = set(nums)
        for i in range(1, len(nums) + 1):
            if i not in nums:
                return i

        return len(nums) + 1

File: leetcode/0041.First-Missing-Positive/FirstMissingPositive.go
package leetcode

// Runtime: 0 ms, faster than 100.00% of Go online submissions for First Missing Positive.
func firstMissingPositive(nums []int) int {
	total := len(nums)
	numMap := make(map[int]bool, total)

	for _, num := range nums {
		if num > 0 {
			numMap[num] = true
		}
	}
	for i := 1; i < total+1; i++ {
		if _, isExist := numMap[i]; !isExist {
			return i
		}
	}
	return total + 1
}
